app/code/Magento/MediaGalleryApi/Model/DataExtractorInterface.php

```diff
@@ -9,6 +9,7 @@
 
 /**
  * Extract data from an object using available getters
+ * @api
```

Comment:
the implementation of this method looks like 
```
    /**
     * Extract data from an object using available getters (does not process extension attributes)
     *
     * @param object $object
     * @param string|null $interface
     * @return array
     */
    public function extract($object, string $interface = null): array;
```

it's semantically incorrect. Should be done like `?string` if it's string|null

Reply:
why do we need this interface here in MediaGallery ? 
` Extract data from an object using available getters`

Reply:
DataExtractor was removed in favour to DataObjectProcessor